packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -482,6 +514,30 @@ export class TreeCheckout implements ITreeCheckoutFork {
 			this.events.emit("commitApplied", data, getRevertible);
 			withinEventContext = false;
 		});
+
+		// When the branch is trimmed, we can garbage collect any repair data whose latest relevant revision is one of the
+		// trimmed revisions.
+		branch.on("ancestryTrimmed", (revisions) => {
+			this.withCombinedVisitor((visitor) => {
+				revisions.forEach((revision) => {
+					this.removedRoots
+						// get all the roots last created or used by the revision
+						.getLatestRelevantRoots(revision)
+						// remove any roots not created by valid revertibles
+						.filter(
+							([_, { major }]) =>
+								major !== undefined && !this.revertibleCommitBranches.has(major),
+						)
```

Comment:
I would have thought that our trunk eviction policy was such that should always be OK to evict this stuff. Does this filter ever filter anything out?

Reply:
It does not, I think I left this in from when I was trying out stuff to get the fuzz tests passing